packages/kbn-optimizer/src/worker/webpack.config.ts

```diff
@@ -135,6 +135,17 @@ export function getWebpackConfig(bundle: Bundle, worker: WorkerConfig) {
                         return null;
                       }
 
+                      if (
```

Comment:
I think we should avoid parsing the directory twice, so we might as well just put this up top:

```ts
if (!parseDirPath(base).dirs.includes('legacy')) {
  return null;
}
```

Reply:
It's a lazy call (because of `&&`), but with your suggestion, we will parse it on every URL. I can restructure code. But it's really performance-critical operation?

Reply:
No, it's not performance critical, I made it sounds that way but it's more an issue because your first instinct in this function was to skip that check. Ideally, moving it up top will prevent others from falling into the same trap. I might not be there to review the code next time this is changed :)

Reply:
done